examples/particle-gibbs/script.jl

```diff
@@ -145,10 +146,13 @@ end
 AdvancedPS.isdone(::NonLinearTimeSeries, step) = step > Tₘ
 
 # We can now sample from the model using the PGAS sampler and collect the trajectories.
-pg = AdvancedPS.PGAS(Nₚ)
-chains = sample(model, pg, Nₛ);
-particles = hcat([trajectory.model.f.X for trajectory in trajectories]...)
-mean_trajectory = mean(particles; dims=2)
+pgas = AdvancedPS.PGAS(Nₚ)
+chains = sample(rng, model, pgas, Nₛ; progress=false);
+trajectories = map(chains) do sample
+    replay(sample.trajectory)
+end
+particles = hcat([trajectory.model.f.X for trajectory in trajectories]...);
```

Comment:
This should have been (without using the suggestion tool, since I messed up everything last time):
```julia
particles = hcat([trajectory.model.f.X for trajectory in chains]...); # pulling it from the chains directly
```
and you should be able to remove the `replay` part

Reply:
I would need to read up on the replay idea here, but replacing 

`trajectories = map(chains) do sample
    replay(sample.trajectory)
end
particles = hcat([trajectory.model.f.X for trajectory in trajectories]...);`

with 

`particles = hcat([trajectory.model.f.X for trajectory in chains]...); # pulling it from the chains directly
`

gives me the error:

ERROR: type PGSample has no field model

Reply:
This fixes this issue
```julia
particles = hcat([chain.trajectory.model.X for chain in pgas_chains]...)
```
but seems to hit another one with the way we copy models around. I'll open an issue and look into it

Reply:
Agree, since the fix pretty much requires https://github.com/TuringLang/AdvancedPS.jl/pull/75 to be merged first